app/models/category_option.rb

```diff
@@ -1,5 +1,10 @@
 class CategoryOption < ActiveRecord::Base
   belongs_to :admin
-  validates :name, presence: true
+  validates :name, presence: true, uniqueness: true
   scope :show_options, -> { order("lower(name) ASC").all }
+
+  def repositories
```

Comment:
I think you can replace this with `has_many :categories`, `has_many :repositories, through: categories`